src/util/exec.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2025 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_EXEC_H
+#define BITCOIN_UTIL_EXEC_H
+
+#include <util/fs.h>
+
+#include <string_view>
+
+namespace util {
+//! Cross-platform wrapper for POSIX execvp function.
+//! Arguments and return value are the same as for POSIX execvp, and the argv
+//! array should consist of null terminated strings and be null terminated
+//! itself, like the POSIX function.
```

Comment:
super nit, but I can't resist :rescue_worker_helmet: :

"null terminated strings" and "null terminated" array -- it is not the same "null":

```suggestion
//! array should consist of '\0'-terminated strings and be nullptr-terminated
//! itself, like the POSIX function.
```
:runner:

Reply:
re: https://github.com/bitcoin/bitcoin/pull/31375#discussion_r2099742506

Yes I used to refer to them as NUL terminated strings, until I came across https://en.wikipedia.org/wiki/Null-terminated_string and decided there wasn't any real ambiguity and null was probably easier to read. But would be ok changing this if others would prefer